src/python/pants/base/payload_field.py

```diff
@@ -139,8 +139,12 @@ def _compute_fingerprint(self):
     hasher.update(self._rel_path)
     for source in sorted(self.relative_to_buildroot()):
       hasher.update(source)
-      with open(os.path.join(get_buildroot(), source), 'rb') as f:
-        hasher.update(f.read())
+      abs_path = os.path.join(get_buildroot(), source)
+      if not os.path.exists(abs_path):
+        hasher.update('\a\n\n\t[FILE DOES NOT EXIST]\t\n\n\a')
+      else:
```

Comment:
protobuf_gen uses sources=from_target, but the code for that explicitly forbids having multiple from_targets, or from_targets combined with "normal" sources.